core/domain/topic_fetchers.py

```diff
@@ -314,3 +314,159 @@ def get_topic_rights_from_model(topic_rights_model):
+
+    Args:
+        topic_ids: list(str). The IDs of topics for which summaries are to be
+            returned.
+
+    Returns:
+        list(TopicSummary). The list of summaries of all given topics present in
+        the datastore.
+    """
+    topic_summaries_models = topic_models.TopicSummaryModel.get_multi(topic_ids)
+    topic_summaries = [
+        get_topic_summary_from_model(summary) if summary else None
+        for summary in topic_summaries_models]
+    return topic_summaries
+
+
+def get_all_skill_ids_assigned_to_some_topic():
+    """Returns the ids of all the skills that are linked to some topics.
+
+    Returns:
+        set([str]). The ids of all the skills linked to some topic.
+    """
+    skill_ids = set([])
+    all_topic_models = topic_models.TopicModel.get_all()
+    all_topics = [get_topic_from_model(topic) for topic in all_topic_models]
+    for topic in all_topics:
+        skill_ids.update(topic.get_all_skill_ids())
+    return skill_ids
+
+
+def get_topic_summary_from_model(topic_summary_model):
+    """Returns a domain object for an Oppia topic summary given a
+    topic summary model.
+
+    Args:
+        topic_summary_model: TopicSummaryModel. The topic summary model object
+            to get the corresponding domain object.
+
+    Returns:
+        TopicSummary. The domain object corresponding to the given model object.
+    """
+    return topic_domain.TopicSummary(
+        topic_summary_model.id, topic_summary_model.name,
+        topic_summary_model.canonical_name,
+        topic_summary_model.language_code,
+        topic_summary_model.description,
+        topic_summary_model.version,
+        topic_summary_model.canonical_story_count,
+        topic_summary_model.additional_story_count,
+        topic_summary_model.uncategorized_skill_count,
+        topic_summary_model.subtopic_count,
+        topic_summary_model.total_skill_count,
+        topic_summary_model.thumbnail_filename,
+        topic_summary_model.thumbnail_bg_color,
+        topic_summary_model.url_fragment,
+        topic_summary_model.topic_model_created_on,
+        topic_summary_model.topic_model_last_updated
+    )
+
+
+def get_topic_summary_by_id(topic_id, strict=True):
+    """Returns a domain object representing a topic summary.
+
+    Args:
+        topic_id: str. ID of the topic summary.
+        strict: bool. Whether to fail noisily if no topic summary with the given
+            id exists in the datastore.
+
+    Returns:
+        TopicSummary or None. The topic summary domain object corresponding to
+        a topic with the given topic_id, if it exists, or else None.
+    """
+    topic_summary_model = topic_models.TopicSummaryModel.get(
+        topic_id, strict=strict)
+    if topic_summary_model:
+        topic_summary = get_topic_summary_from_model(topic_summary_model)
+        return topic_summary
+    else:
+        return None
+
+
+def get_new_topic_id():
```

Comment:
Not sure if this should be a topic_fetchers function, since this is actually only used when creating a topic, to get a new topic id. I'm fine with having it either way, though what do you think?

Reply:
@aks681 Thanks for the suggestion,
to me, this sounds like a getter function (as it starts with get_*) so I placed it here